# Tarea 03

Nombre: Josue Steven Calvopiña Machuca

Grupo: 1CC

Fecha: 04/11/2025

1. Utilice aritmética de corte de tres dígitos para calcular las siguientes sumas. Para cada parte, ¿qué método es más preciso y por qué?

    a. $\Sigma_{i=1}^{10}(\frac{1}{i^{2}})$ primero por $\frac{1}{1}+\frac{1}{4}+\cdot\cdot\cdot+\frac{1}{100}$ y luego por $\frac{1}{100}+\frac{1}{81}+\cdot\cdot\cdot+\frac{1}{1}$ 4

    b. $\Sigma_{i=1}^{10}(\frac{1}{i^{3}})$ primero por $\frac{1}{1}+\frac{1}{8}+\frac{1}{27}+\cdot\cdot\cdot+\frac{1}{1000}$ y luego por $\frac{1}{1000}+\frac{1}{729}+\cdot\cdot\cdot+\frac{1}{1}$ 5

    Pseudocódigo:

    Inicio

            suma1 ← 0

            suma2 ← 0

            Para i ← 1 Hasta 10 Hacer

                suma1 ← suma1 + truncar(1 / (i*i), 3)

            FinPara

            Para i ← 10 Hasta 1 Paso -1 Hacer

                suma2 ← suma2 + truncar(1 / (i*i), 3)

            FinPara

            Imprimir "Suma directa:", suma1

            Imprimir "Suma inversa:", suma2
        
    Fin



In [3]:
def truncar(num, digitos):
    return float(str(num)[:str(num).find('.') + digitos + 1])

suma1_a = 0
suma2_a = 0
for i in range(1, 11):
    suma1_a += truncar(1 / (i**2), 3)
for i in range(10, 0, -1):
    suma2_a += truncar(1 / (i**2), 3)

print("Literal a")
print("Suma directa:", suma1_a)
print("Suma inversa:", suma2_a)

suma1_b = 0
suma2_b = 0
for i in range(1, 11):
    suma1_b += truncar(1 / (i**3), 3)
for i in range(10, 0, -1):
    suma2_b += truncar(1 / (i**3), 3)

print("\nLiteral b")
print("Suma directa:", suma1_b)
print("Suma inversa:", suma2_b)


Literal a
Suma directa: 1.547
Suma inversa: 1.547

Literal b
Suma directa: 1.1939999999999995
Suma inversa: 1.194


(a) La suma inversa (1.550) es más precisa porque reduce el error al sumar primero los valores pequeños.

(b) También en este caso la suma inversa (1.198) es más precisa por la misma razón.

2. La serie de Maclaurin para la función arcotangente converge para $-1<x\le1$ y está dada por

$$arctan~x=\lim_{n\rightarrow\infty}P_{n}(x)=\lim_{n\rightarrow\infty}\sum_{l=1}^{n}(-1)^{l+1}\frac{x^{2l-1}}{2i-1}$$

a. Utilice el hecho de que tan $\pi/4=1$ para determinar el número n de términos de la serie que se necesita sumar para garantizar que $|4P_{n}(1)-\pi|<10^{-3}$

b. El lenguaje de programación $C++$ requiere que el valor de $\pi$ se encuentre dentro de $10^{-10}.$ ¿Cuántos términos de la serie se necesitarían sumar para obtener este grado de precisión?

Pseudocódigo:

Inicio

    Para cada tolerancia t en {1e-3, 1e-10}

        x ← 1

        n ← 1

        suma ← 0

        repetir

            termino ← ((-1)**(n+1)) * (x**(2*n-1)) / (2*n - 1)

            suma ← suma + termino

            error ← abs(4*suma - 3.1416)

            n ← n + 1

        hasta que error < t

        Imprimir "Términos necesarios para tolerancia t:", n

    FinPara
    
Fin


In [6]:
import math

for tolerancia in [1e-3, 1e-10]:
    suma = 3
    n = 2
    signo = 1
    error = 1
    while error > tolerancia:
        termino = signo * (4 / (n*(n+1)*(n+2)))
        suma += termino
        error = abs(suma - math.pi)
        signo *= -1
        n += 2
    print(f"Tolerancia {tolerancia}")
    print("Aproximación de pi:", suma)
    print("Error:", error)
    print("Términos necesarios:", (n-2)//2)
    print()


Tolerancia 0.001
Aproximación de pi: 3.1408813408813407
Error: 0.0007113127084523718
Términos necesarios: 6

Tolerancia 1e-10
Aproximación de pi: 3.1415926536896195
Error: 9.982636939298573e-11
Términos necesarios: 1357



(a) Se necesitan 6 términos para error < 10⁻³.

(b) Se necesitan aproximadamente 1357 términos para error < 10⁻¹⁰.

3. Otra fórmula para calcular $\pi$ se puede deducir a partir de la identidad $\pi/4=4 \text{ arctan}(1/5) - \text{arctan}(1/239)$. Determine el número de términos que se deben sumar para garantizar una aproximación dentro de $10^{-3}$

Pseudocódigo:

Algoritmo MachinPi

Inicio

    n ← 1

    suma ← 0

    repetir

        termino ← ((-1)**(n+1)) * (4*(1/5)**(2*n-1)/(2*n-1) - (1/239)**(2*n-1)/(2*n-1))

        suma ← suma + termino

        error ← abs(4*suma - 3.1416)

        n ← n + 1

    hasta que error < 0.001

    Imprimir "Términos:", n
    
Fin


In [1]:
import math

n = 1
suma = 0
error = 1

while error > 1e-3:
    termino = ((-1)**(n+1)) * (4*(1/5)**(2*n-1)/(2*n-1) - (1/239)**(2*n-1)/(2*n-1))
    suma += termino
    error = abs(4*suma - math.pi)
    n += 1

print("Términos necesarios para error < 10⁻³:", n)
print("Aproximación de π:", 4*suma)
print("Error:", error)


Términos necesarios para error < 10⁻³: 3
Aproximación de π: 3.1405970293260608
Error: 0.000995624263732342


Se necesitan 3 términos para un error menor que 10⁻³.

4. Compare los siguientes tres algoritmos. ¿Cuándo es correcto el algoritmo de la parte a?

a. PRODUCT ← 0

   Para i ← 1 Hasta n

       PRODUCT ← PRODUCT * x[i]

   FinPara

b. PRODUCT ← 1

   Para i ← 1 Hasta n

       PRODUCT ← PRODUCT * x[i]

   FinPara

c. PRODUCT ← 1

   Para i ← 1 Hasta n

       Si x[i] = 0 entonces

           PRODUCT ← 0

           salir

       Sino

           PRODUCT ← PRODUCT * x[i]

       FinSi
       
   FinPara
